### DISCUSSION TITLE
library swipe javascript

### DISCUSSION THREAD

**Discussion by itsleviaa10** [community/community#146404] [Programming Help]
### Body

I want to ask, A library for javascript swipe animation.

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by pooulad** [comment]
For JavaScript swipe animations, here are some popular libraries you can use:

### 1\. [**Swiper.js**](https://swiperjs.com/)

*   **Features**:
    
    *   Highly customizable and modern touch slider/swiper.
        
    *   Supports free mode, responsive design, and various effects like fade, cube, and cover flow.
        
*   **Use Case**: Ideal for creating swipe-based sliders and carousels.
    
*   bashCopy codenpm install swiper
    
*   **Documentation**: Available on the [Swiper.js website](https://swiperjs.com/).
    

### 2\. **Hammer.js**

*   **Features**:
    
    *   A lightweight library for recognizing gestures like swipe, pinch, and drag.
        
    *   Integrates well with other animation libraries.
        
*   **Use Case**: Perfect for custom gesture handling, including swipe animations.
    
*   bashCopy codenpm install hammerjs
    

### 3\. [**Anime.js**](https://animejs.com/)

*   **Features**:
    
    *   General-purpose animation library with swipe support via JavaScript events.
        
    *   Excellent for fine-tuned animations and transitions.
        
*   **Use Case**: Useful when swipe gestures are part of a larger animation context.
    
*   bashCopy codenpm install animejs
    

### 4\. **GSAP (GreenSock Animation Platform)**

*   **Features**:
    
    *   Powerful animation platform supporting swipe gestures via event listeners.
        
    *   Works seamlessly with touch gestures for complex animations.
        
*   **Use Case**: Great for highly interactive UIs and advanced animations.
    
*   bashCopy codenpm install gsap
    

### 5\. [**React Swipeable**](https://www.npmjs.com/package/react-swipeable)

*   **Features**:
    
    *   Built specifically for React projects.
        
    *   Simplifies swipe event handling and can trigger animations or navigation.
        
*   **Use Case**: If you're building with React and need swipe animations.
    
*   bashCopy codenpm install react-swipeable

**Reply by mecodeatlas** [reply]
Hey @ATHAANGGORO10! Did @pooulad's response help you with your question?

**Reply by itsleviaa10** [reply]
Thank you, Senior @pooulad.👌
